services/ymax-planner/test/plan-withdraw.property.test.ts

```diff
@@ -0,0 +1,214 @@
+import { testProp, fc } from '@fast-check/ava';
+import test from 'ava';
+
+import { AmountMath } from '@agoric/ertp';
+import type { Brand } from '@agoric/ertp/src/types.js';
+import { Far } from '@endo/pass-style';
+import {
+  computeWeightedTargets,
+  planWithdrawFromAllocations,
+} from '../src/plan-deposit.ts';
+import { mockGasEstimator } from './mocks.ts';
+
+const depositBrand = Far('mock brand') as Brand<'nat'>;
+const feeBrand = Far('fee brand (BLD)') as Brand<'nat'>;
+const makeDeposit = (value: bigint) => AmountMath.make(depositBrand, value);
+
+const withdrawCase = fc
+  .record({
+    USDN: fc.nat({ max: 5_000 }),
+    Aave_Arbitrum: fc.nat({ max: 5_000 }),
+    Compound_Base: fc.nat({ max: 5_000 }),
+  })
+  .filter(record => Object.values(record).some(v => v > 0))
+  .chain(record => {
+    const total = Object.values(record).reduce((sum, n) => sum + n, 0);
+    return fc.record({
+      balances: fc.constant(record),
+      withdrawal: fc.integer({ min: 0, max: total }),
+    });
+  });
+
+testProp(
+  'computeWeightedTargets distributes withdraw proportionally and keeps the remainder on the heaviest weight',
```

Comment:
This is a good concept, but is not being tested correctly. I think the appropriate input would be balances of dynamic cardinality and minimum amount of e.g. 1000 uusdc and appropriate assertions would be that the relative share after withdrawal shifts by no more than e.g. 1%.